common/src/FileLogger.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ Copyright (C) 2010-2016 Kristian Duske
+ 
+ This file is part of TrenchBroom.
+ 
+ TrenchBroom is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+ 
+ TrenchBroom is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+ 
+ You should have received a copy of the GNU General Public License
+ along with TrenchBroom. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "FileLogger.h"
+
+#include "IO/Path.h"
+#include "IO/SystemPaths.h"
+
+#include <cassert>
+
+#include <wx/string.h>
+
+namespace TrenchBroom {
+    FileLogger::FileLogger(const IO::Path& filePath) :
+    m_file(std::fopen(filePath.asString().c_str(), "w")) {}
+    
+    FileLogger::~FileLogger() {
+        if (m_file != NULL) {
+            fclose(m_file);
+            m_file = NULL;
+        }
+    }
+    
+    FileLogger& FileLogger::instance() {
+        static FileLogger Instance(IO::SystemPaths::logFilePath());
```

Comment:
I don't see where the parent directory of `IO::SystemPaths::logFilePath()` is created, is that a potential problem?

Testing on macOS, I tried deleting `~/Library/Application Support/TrenchBroom` and then launching TB, and the `TrenchBroom` directory was created and everything worked.